apps/frontend/src/components/map-navigation/where-am-i.tsx

```diff
@@ -1,11 +1,90 @@
-import {Button} from '@/components/ui/button';
+import { Button } from '@/components/ui/button';
+import { useGeolocated } from "react-geolocated";
+import { useState, useEffect } from 'react';
+import { LatLngTuple } from 'leaflet';
+
+// recalculations for the patchname
+// Constants for the grid
+const GRID_SIZE_FEET = 15;
+const EARTH_RADIUS_FEET = 20902231; // Earth's radius in feet
+const CENTER_LAT = 42.04937; // Our center latitude
+
+// Calculate accurate conversion factors
+const FEET_TO_DEGREES_LAT = (1 / EARTH_RADIUS_FEET) * (180 / Math.PI); // For latitude
+const FEET_TO_DEGREES_LNG =
+  (1 / (EARTH_RADIUS_FEET * Math.cos((CENTER_LAT * Math.PI) / 180))) * (180 / Math.PI); // For longitude
+
+// Grid boundaries
+const TOP_LEFT: LatLngTuple = [42.0500576, -87.6739709];
+const BOTTOM_RIGHT: LatLngTuple = [42.0485898, -87.6729095];
+const CENTER: LatLngTuple = [42.04937, -87.673388];
```

Comment:
Put all of these in a `constants.tsx` and then export them to this file and the map view file